blocks/extensions/__init__.py

```diff
@@ -141,36 +143,39 @@ class SimpleExtension(TrainingExtension):
         If not ``None``, :meth:`do` is invoked after every n-th batch.
 
     """
-    def __init__(self, before_training=False, before_first_epoch=False,
-                 on_resumption=False, on_interrupt=False,
-                 after_every_epoch=False, after_every_batch=False,
-                 after_training=False,
-                 after_n_epochs=None, after_n_batches=None,
-                 every_n_batches=None, **kwargs):
-        super(SimpleExtension, self).__init__(**kwargs)
+    BOOLEAN_TRIGGERS = frozenset(["before_training", "before_first_epoch",
+                                  "on_resumption", "on_interrupt",
+                                  "after_every_epoch", "after_every_batch",
+                                  "after_training"])
+
+    INTEGER_TRIGGERS = frozenset(["after_n_epochs", "after_n_batches",
+                                  "every_n_epochs", "every_n_batches"])
+
+    def __init__(self, **kwargs):
         self._conditions = []
-        if before_training:
-            self.add_condition("before_training")
-        if before_first_epoch:
-            self.add_condition(
-                "before_epoch",
-                predicate=lambda log: log.status.epochs_done == 0)
-        if on_resumption:
-            self.add_condition("on_resumption")
-        if on_interrupt:
-            self.add_condition("on_interrupt")
-        if after_every_epoch:
-            self.add_condition("after_epoch")
-        if after_every_batch:
-            self.add_condition("after_batch")
-        if after_training:
-            self.add_condition("after_training")
-        if after_n_epochs:
-            self.invoke_after_n_epochs(after_n_epochs)
-        if after_n_batches:
-            self.invoke_after_n_batches(after_n_batches)
-        if every_n_batches:
-            self.invoke_every_n_batches(every_n_batches)
+        predicates = {"before_epoch": lambda log: log.status.epochs_done == 0}
+        predicate_factories = {
+            'every_n_batches': lambda n_batches:
+                lambda log: log.status.iterations_done % n_batches == 0,
+            'after_n_batches': lambda n_batches:
+                lambda log: log.status.iterations_done == n_batches,
+            'every_n_epochs': lambda n_epochs:
+                lambda log: log.status.epochs_done % n_epochs == 0,
+            'after_n_epochs': lambda n_epochs:
+                lambda log: log.status.epochs_done == n_epochs,
+        }
+        # Freeze the keys as a list so that we can safely modify kwargs.
+        for key in list(kwargs.keys()):
+            if key in self.BOOLEAN_TRIGGERS and kwargs[key]:
+                self.add_condition(key, predicate=predicates.get(key, None))
```

Comment:
Same here, there is no condition `after_every_batch`, the condition here is `after_batch`. In general, the keyword arguments and the conditions don't always match.